### DISCUSSION TITLE
Sending CC on all channels

### DISCUSSION THREAD

**Discussion by KeenVox** [tttapa/Control-Surface#629] [Q&A]
Hi,
I'm building a midi controller on a teensy lc mostly out of pots to send cc messages. I want to send the messages on all channels simultaneously. 
Do I gave to copy the code line for the every pot for ervery channel or is there a command to tell it to send to every channel at once?
Thx!

**Comment by tttapa** [comment]
MIDI doesn't support sending a message to all channels at the same time. You'll have to send 16 separate messages. You can implement your own version of the ContinuousCC MIDI sender:

https://github.com/tttapa/Control-Surface/blob/e6542665da8997fdf33aee5ff1e0b7383c7ebe00/src/MIDI_Senders/ContinuousCCSender.hpp#L14-L23

Use a for loop to send on all 16 channels.

Then use this MIDI sender to create a custom CCPotentiometer:

https://github.com/tttapa/Control-Surface/blob/e6542665da8997fdf33aee5ff1e0b7383c7ebe00/src/MIDI_Outputs/CCPotentiometer.hpp#L19-L34
